packages/rocketchat-integrations/server/api/api.js

```diff
@@ -0,0 +1,358 @@
+				return {
+					result: HTTP.call(method, url, options)
+				};
+			} catch (error) {
+				return {
+					error
+				};
+			}
+		}
+	};
+
+	Object.keys(RocketChat.models).filter(function(k) {
+		return !k.startsWith('_');
+	}).forEach((k) => {
+		return sandbox[k] = RocketChat.models[k];
+	});
+	return { store, sandbox	};
+};
+
+const getIntegrationScript = function(integration) {
+	const compiledScript = compiledScripts[integration._id];
+	if ((compiledScript != null) && +compiledScript._updatedAt === +integration._updatedAt) {
+		return compiledScript.script;
+	}
+	const script = integration.scriptCompiled;
+	const sandboxItems = buildSandbox();
+	try {
+		logger.incoming.info('Will evaluate script of Trigger', integration.name);
+		logger.incoming.debug(script);
+		const vmScript = vm.createScript(script, 'script.js');
+		vmScript.runInNewContext(sandboxItems.sandbox);
+		if (sandboxItems.sandbox.Script != null) {
+			compiledScripts[integration._id] = {
+				script: new sandboxItems.sandbox.Script(),
+				store: sandboxItems.store,
+				_updatedAt: integration._updatedAt
+			};
+			return compiledScripts[integration._id].script;
+		}
+	} catch ({stack}) {
+		logger.incoming.error('[Error evaluating Script in Trigger', integration.name, ':]');
+		logger.incoming.error(script.replace(/^/gm, '  '));
+		logger.incoming.error('[Stack:]');
+		logger.incoming.error(stack.replace(/^/gm, '  '));
+		throw RocketChat.API.v1.failure('error-evaluating-script');
+	}
+	if (sandboxItems.sandbox.Script == null) {
+		logger.incoming.error('[Class "Script" not in Trigger', integration.name, ']');
+		throw RocketChat.API.v1.failure('class-script-not-found');
+	}
+};
+
+Api = new Restivus({
+	enableCors: true,
+	apiPath: 'hooks/',
+	auth: {
+		user() {
+			const payloadKeys = Object.keys(this.bodyParams);
+			const payloadIsWrapped = (this.bodyParams && this.bodyParams.payload) && payloadKeys.length === 1;
+			if (payloadIsWrapped && this.request.headers['content-type'] === 'application/x-www-form-urlencoded') {
+				try {
+					this.bodyParams = JSON.parse(this.bodyParams.payload);
+				} catch ({message}) {
+					return {
+						error: {
+							statusCode: 400,
+							body: {
+								success: false,
+								error: message
+							}
+						}
+					};
+				}
+			}
+			this.integration = RocketChat.models.Integrations.findOne({
+				_id: this.request.params.integrationId,
+				token: decodeURIComponent(this.request.params.token)
+			});
+			if (this.integration == null) {
+				logger.incoming.info('Invalid integration id', this.request.params.integrationId, 'or token', this.request.params.token);
+				return;
+			}
+			const user = RocketChat.models.Users.findOne({
+				_id: this.integration.userId
+			});
+			return {user};
+		}
+	}
+});
+
+const createIntegration = function(options, user) {
+	logger.incoming.info('Add integration', options.name);
+	logger.incoming.debug(options);
+	Meteor.runAsUser(user._id, function() {
+		switch (options['event']) {
```

Comment:
Convert this to `options.event` so the assessors are like the other ones for options.